crates/sui-json-rpc/src/name_service.rs

```diff
@@ -278,6 +278,11 @@ fn validate_label(label: &str) -> Result<&str, NameServiceError> {
             _ => false,
         };
 
+        // Do not allow consecutive dashes (--)
```

Comment:
Mostly for my education -- why is this an issue?

Reply:
Oof, thanks for flagging this. It's actually a mistake that exists on our other places (SDKs, Frontend), rather than something that should be enforceable here. Confirmation bias hit me.

Only dashes (`-`) in the beginning & ending seem to be invalid (and that's the only check that happens on-chain too).